eng/common/TestResources/New-TestResources.ps1

```diff
@@ -289,7 +289,7 @@ try {
                 $AzureTestPrincipal
             } else {
                 Log "TestApplicationId was not specified; creating a new service principal in subscription '$SubscriptionId'"
-                $global:AzureTestPrincipal = New-AzADServicePrincipal -Role Owner -Scope "/subscriptions/$SubscriptionId"
+                $global:AzureTestPrincipal = New-AzADServicePrincipal -Role Owner -Scope "/subscriptions/$SubscriptionId" -DisplayName "test-resources-$($baseName).microsoft.com"
```

Comment:
@benbp Will this display name cause problems in other clouds that aren't using the microsoft public tenant?

Reply:
I don't think it will cause functional problems, as it's just a display name, but it would make sense to see if we can grab the suffix out of the tenant/authority host info in the context and use that instead (or just not have the display name include `.com`).